src/cpp/src/sampler.hpp

```diff
@@ -42,6 +43,21 @@ struct SamplerOutput {
     size_t num_generated_tokens = 0;
 };
 
+struct AssistingPipelineInfo {
+    size_t max_removed_tokens_per_request = 0; 
+    size_t min_generated_len = std::numeric_limits<size_t>::max();
+    size_t updated_validation_len = 0;
+};
+
+struct SequenceGroupSamplingInfo {
+    SamplerOutput sampler_output;
+    AssistingPipelineInfo assisting_pipeline_info;
+
+    AssistingPipelineInfo& get_assisting_pipeline_info() {
+        return assisting_pipeline_info;
+    }
+};
```

Comment:
looks like these structures are used only within .cpp file => let's move it there

Reply:
I use `AssistingPipelineInfo` full type in `SequenceGroupSamplingInfo` and `sample_from_sequence_group` method refers this type in .hpp file.